tests/conftest.py

```diff
@@ -1,80 +1,251 @@
+        restart=False,
+        shutdown=TEARDOWN,
+        null_server=config.option.no_server,
+        verbose=VERBOSE,
     )
 
-    markexpr_list = []
 
-    if config.option.current:
-        markexpr_list.append("current")
+# ------------------- FIXTURES -------------------
+
+
+@pytest.fixture(scope="session")
+def verbose():
+    """Config for pytest fixtures."""
+    yield VERBOSE
+
+
+@pytest.fixture(scope="session", autouse=True)
+def verbose_context(verbose):
+    """Verbosity context for suppressing Spyglass logging."""
+    yield nullcontext() if verbose else QuietStdOut()
+
+
+@pytest.fixture(scope="session")
+def teardown(request):
+    yield TEARDOWN
+
+
+@pytest.fixture(scope="session")
+def server(request, teardown):
+    SERVER.wait()
+    yield SERVER
+    if teardown:
+        SERVER.stop()
+
+
+@pytest.fixture(scope="session")
+def dj_conn(request, server, verbose, teardown):
+    """Fixture for datajoint connection."""
+    config_file = "dj_local_conf.json_pytest"
+
+    dj.config.update(server.creds)
+    dj.config["loglevel"] = "INFO" if verbose else "ERROR"
+    dj.config.save(config_file)
+    dj.conn()
+    yield dj.conn()
+    if teardown:
+        if Path(config_file).exists():
+            os.remove(config_file)
+
+
+@pytest.fixture(scope="session")
+def base_dir():
+    yield BASE_DIR
+
+
+@pytest.fixture(scope="session")
+def raw_dir(base_dir):
+    # could do settings.raw_dir, but this is faster while server booting
+    yield base_dir / "raw"
+
+
+@pytest.fixture(scope="session")
+def minirec_path(raw_dir):
+    yield raw_dir / "test.nwb"
+
+
+@pytest.fixture(scope="session")
+def minirec_download():
+    # test_path = (
+    #     "ipfs://bafybeie4svt3paz5vr7cw7mkgibutbtbzyab4s24hqn5pzim3sgg56m3n4"
+    # )
+    # try:
+    #     local_test_path = kcl.load_file(test_path)
+    # except Exception as e:
+    #     if os.environ.get("KACHERY_CLOUD_EPHEMERAL", None) != "TRUE":
+    #         print(
+    #             "Cannot load test file in non-ephemeral mode. Kachery cloud"
+    #             + "client may need to be registered."
+    #         )
+    #     raise e
+    # os.rename(local_test_path, nwbfile_path)
+    pass
+
+
+@pytest.fixture(scope="session")
+def minirec_content(minirec_path):
+    with pynwb.NWBHDF5IO(
+        path=str(minirec_path), mode="r", load_namespaces=True
+    ) as io:
+        nwbfile = io.read()
+        yield nwbfile
+
+
+@pytest.fixture(scope="session")
+def minirec_open(minirec_content):
+    yield minirec_content
+
+
+@pytest.fixture(scope="session")
+def minirec_closed(minirec_path):
+    with pynwb.NWBHDF5IO(
+        path=str(minirec_path), mode="r", load_namespaces=True
+    ) as io:
+        nwbfile = io.read()
+    yield nwbfile
+
+
+@pytest.fixture(scope="session")
+def minirec_devices(minirec_content):
+    yield minirec_content.devices
+
+
+@pytest.fixture(scope="session")
+def minirec_insert(minirec_path, teardown, server, dj_conn):
```

Comment:
Delay import until after dj_conn fixture to avoid issue with undeclared config/base_dir